### ISSUE TITLE
Cached tasks do not have names reassigned with task_run_name

### ISSUE THREAD

**Issue by PiotrSiejda** [PrefectHQ/prefect#8480]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Prefect documentation for this issue.
- [X] I checked that this issue is related to Prefect and not one of its dependencies.

### Bug summary

Recently concept of task_run_name was introduced in:
https://github.com/PrefectHQ/prefect/pull/8378

Unfortunately it does not seem to work for cached task_runs

To reproduce please run twice code below:

### Reproduction

```python3
from prefect import flow, task
from prefect.tasks import task_input_hash
import datetime


@task(task_run_name="Task_for_{in_str}", cache_key_fn=task_input_hash, cache_expiration=datetime.timedelta(minutes=10))
def task1(in_str: str):
    return None


@flow(flow_run_name="My_flow_name2")
def ff():
    ss = ["ww", "aa", "pp"]
    task1.map(ss)


if __name__ == "__main__":
    ff()
```

### Error

```python3
Second run tasks are not renamed.
```

### Versions

```Text
Version:             2.8.0
API version:         0.8.4
Python version:      3.9.13
Git commit:          50f0ad0a
Built:               Thu, Feb 9, 2023 3:38 PM
OS/Arch:             win32/AMD64
Profile:             default
Server type:         ephemeral
Server:
  Database:          sqlite
  SQLite version:    3.37.2
```

**Comment by serinamarie** [comment]
Hi @PiotrSiejda, we appreciate you raising this issue 🙂  When a new task run is created like your MRE above, it will have a state of CACHED, and since we update the task run name inside of a while RUNNING block, the task name does not get updated:
https://github.com/PrefectHQ/prefect/blob/671adf736be5b65e6568dd1a042c97902d5b48f0/src/prefect/engine.py#L1480-L1491

**Comment by cicdw** [comment]
This behavior has been fixed in 3.0: regardless of whether a task is cached or not, it's run will be properly renamed.
